R/guides.R

```diff
@@ -134,28 +129,46 @@ complete_guide_theme <- function(theme) {
   }
   theme
 }
-#' @importFrom grid valid.just
-assemble_guides <- function(guides, theme) {
-  theme <- complete_guide_theme(theme)
+#' @importFrom utils getFromNamespace
+#' @importFrom ggplot2 calc_element
+assemble_guides <- function(guides, position, theme) {
+  # https://github.com/tidyverse/ggplot2/blob/57ba97fa04dadc6fd73db1904e39a09d57a4fcbe/R/guides-.R#L512
+  theme$legend.spacing <- theme$legend.spacing %||% unit(0.5, "lines")
+  theme$legend.spacing.y <- calc_element("legend.spacing.y", theme)
+  theme$legend.spacing.x <- calc_element("legend.spacing.x", theme)
+
+  # for every position, collect all individual guides and arrange them
+  # into a guide box which will be inserted into the main gtable
+  Guides <- getFromNamespace("Guides", "ggplot2")
```

Comment:
This is the only part that makes me pause. This will fail for older versions of ggplot2 and I would like to support those. Could you make it so that your fix only applies to versions of ggplot2 with the new guide system and falls back to the current behaviour for older ggplot2 versions? I think the new guide system coincided with the option for null sizing of guides anyway